gtk4/src/subclass/native_dialog.rs

```diff
@@ -0,0 +1,108 @@
+use gtk_sys;
+
+use glib::subclass::object::ObjectImpl;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, Object};
+
+use NativeDialog;
+use ResponseType;
+
+pub trait NativeDialogImpl: NativeDialogImplExt + ObjectImpl {
+    fn response(&self, dialog: &Self::Type, response: ResponseType) {
+        self.parent_response(dialog, response)
+    }
+
+    fn show(&self, dialog: &Self::Type) {
+        self.parent_show(dialog)
+    }
+
+    fn hide(&self, dialog: &Self::Type) {
+        self.parent_hide(dialog)
+    }
+}
+
+pub trait NativeDialogImplExt: ObjectSubclass {
+    fn parent_response(&self, dialog: &Self::Type, response: ResponseType);
+    fn parent_show(&self, dialog: &Self::Type);
+    fn parent_hide(&self, dialog: &Self::Type);
+}
+
+impl<T: NativeDialogImpl> NativeDialogImplExt for T {
+    fn parent_response(&self, dialog: &Self::Type, response: ResponseType) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkNativeDialogClass;
+            if let Some(f) = (*parent_class).response {
+                f(
+                    dialog.unsafe_cast_ref::<NativeDialog>().to_glib_none().0,
+                    response.to_glib(),
+                )
+            }
+        }
+    }
+
+    fn parent_show(&self, dialog: &Self::Type) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkNativeDialogClass;
+            let f = (*parent_class)
+                .show
+                .expect("No parent class impl for \"show\"");
```

Comment:
Just to make sure I understood how things works properly, this can't be `NULL` per https://github.com/GNOME/gtk/blob/master/gtk/gtknativedialog.c#L308 ?

Reply:
Yes, that's correct

Reply:
You could also do nothing here though, it doesn't matter much because you don't have to invent a return value out of thin air

Reply:
Right, thanks!